Source/Libraries/GSF.Security/AdoSecurityProvider.cs

```diff
@@ -574,7 +574,30 @@ public override bool RefreshData()
 
                 // Filter explicitly assigned application roles for current user - this will return an empty set if no
                 // explicitly defined roles exist for the user -or- user doesn't exist in the database.
-                DataRow[] userApplicationRoles = securityContext.Tables[ApplicationRoleUserAccountTable].Select($"UserAccountID = '{EncodeEscapeSequences(userAccountID.ToString())}'");
+                DataRow[] userApplicationRoles = securityContext.Tables[ApplicationRoleUserAccountTable]
+                    .Select($"UserAccountID = '{EncodeEscapeSequences(userAccountID.ToString())}'");
+
+                //Filter ApplicationRoles by NodeID in case multiple Nodes use the same database
+                Func<DataRow, bool> nodeIDFilter = (row) =>
+                {
+                    DataRow applicationRole = null;
+
+                    if (securityContext.Tables[ApplicationRoleTable].PrimaryKey.Length > 0)
+                        applicationRole = securityContext.Tables[ApplicationRoleTable].Rows.Find(row["ApplicationRoleID"]);
```

Comment:
Nit-picky thing: Add braces around this initial `if` result, our coding style spec says that if you use braces around one section, you should use it around the other.